opm/simulators/wells/MultisegmentWellEval.cpp

```diff
@@ -884,9 +884,14 @@ pressureDropSpiralICD(const int seg) const
         density.clearDerivatives();
     }
 
-    const EvalWell liquid_emulsion_viscosity = mswellhelpers::emulsionViscosity(water_fraction, water_viscosity,
-                                                                                oil_fraction, oil_viscosity, sicd);
-    const EvalWell mixture_viscosity = (water_fraction + oil_fraction) * liquid_emulsion_viscosity + gas_fraction * gas_viscosity;
+
+    const EvalWell liquid_fraction = water_fraction + oil_fraction;
+
+    // viscosity contribution from the liquid
+    const EvalWell liquid_viscosity_fraction = liquid_fraction < 1.e-30 ? oil_fraction * oil_viscosity + water_fraction * water_viscosity :
+            liquid_fraction * mswellhelpers::emulsionViscosity(water_fraction, water_viscosity, oil_fraction, oil_viscosity, sicd);
```

Comment:
1e-30 is a very small number for a fraction. Could it be more robust to use a larger number i.e. 1e-12?

Reply:
This PR is only for removing the singularity warning and trying to minimizing the change of the results unnecessarily as much as possible.  The problematic values the introduces singular matrices problems are on the e-160 level. So whether to use 1.e-12 or 1.e-30 should be a optimization problem. To me, they should be similar.